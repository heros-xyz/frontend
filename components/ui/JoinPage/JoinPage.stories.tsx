import type { Meta, StoryObj } from "@storybook/react";
import JoinPage from "./index";

const meta: Meta<typeof JoinPage> = {
  title: "Components/JoinPage",
  component: JoinPage,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof JoinPage>;

export const JoinHerosPage: Story = {};
