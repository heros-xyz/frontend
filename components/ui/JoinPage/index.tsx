import { Box, Button, Center, Text } from "@chakra-ui/react";
import Link from "next/link";
import { Logo } from "@/components/svg/Logo";

const JoinPage = () => {
  return (
    <Box display="flex" justifyContent="center" bg="primary" height="100vh">
      <Center>
        <Box px={5}>
          <Logo
            w={{ base: "161px", xl: "323px" }}
            height={{ base: "32px", xl: "64px" }}
          />

          <Box mt={{ base: "12", xl: "20" }}>
            <Text
              fontSize={{ base: "md", xl: "xl" }}
              color="white"
              pb={{ base: "2.5", xl: "6" }}
            >
              Are you joining as a fan or an athlete?
            </Text>
            <Box color="secondary" mt={4}>
              <Link href="/fan/sign-up">
                <Button
                  w="100%"
                  h="inherit"
                  variant="secondaryOutline"
                  justifyContent="flex-start"
                  px="5"
                  py={{ base: "2", xl: "3" }}
                >
                  Fan
                </Button>
              </Link>
              <Text
                fontSize={{ base: "xs", xl: "md" }}
                pt={{ base: "1", xl: "3.5" }}
                pb={{ base: "2.5", xl: "7" }}
              >
                As a fan you can follow & support your favorite athlete(s)
                directly.
              </Text>
              <Link href="/athlete/sign-up">
                <Button
                  width="100%"
                  h="inherit"
                  variant="secondaryOutline"
                  justifyContent="flex-start"
                  px="5"
                  py={{ base: "2", xl: "3" }}
                >
                  Athlete
                </Button>
              </Link>
              <Text
                fontSize={{ base: "xs", xl: "md" }}
                pt={{ base: "1", xl: "3.5" }}
              >
                As an athlete you can share your stories, thoughts,... with your
                fans and earn money while doing so.
              </Text>
            </Box>
          </Box>
        </Box>
      </Center>
    </Box>
  );
};

export default JoinPage;
