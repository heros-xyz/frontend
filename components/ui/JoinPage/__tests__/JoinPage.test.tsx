import { render } from "@testing-library/react";
import JoinPage from "../index";

it("renders homepage unchanged", () => {
  const { container } = render(<JoinPage />);
  expect(container).toMatchSnapshot();
});
