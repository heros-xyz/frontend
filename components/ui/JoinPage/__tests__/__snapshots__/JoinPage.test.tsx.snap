// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="css-1j7etba"
  >
    <div
      class="css-12h75we"
    >
      <div
        class="css-10x2qr6"
      >
        <svg
          class="chakra-icon css-1ju7ach"
          focusable="false"
          viewBox="0 0 154 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clipPath="url(#clip0_7538_85644)"
          >
            <path
              d="M43.8695 24.8876C43.2568 25.1525 42.6786 25.4975 42.0596 25.7439C40.7618 26.2584 39.398 26.3292 38.0185 26.1875C35.8755 25.9657 34.4771 24.7767 33.9524 22.7281C33.8769 22.4355 33.8235 22.1336 33.7575 21.8132C39.4577 21.7824 45.1013 21.7516 50.7606 21.7208C50.7606 20.9753 50.7669 20.2822 50.7606 19.5891C50.7481 18.2399 50.5721 16.9122 50.1825 15.6184C49.1958 12.3377 47.1815 9.91951 43.9041 8.62571C41.4279 7.6492 38.845 7.46437 36.24 7.96341C32.3498 8.70889 29.528 10.8529 27.9317 14.4263C26.6527 17.288 26.4956 20.2699 27.2466 23.2949C28.3873 27.891 32.1329 31.1902 36.9219 31.8464C37.3146 31.9018 37.7106 31.9511 38.1034 32.0035H39.9511C40.0202 31.9881 40.0862 31.9603 40.1584 31.9573C41.5128 31.871 42.8388 31.6338 44.1178 31.1841C46.5342 30.34 48.495 28.9353 49.7802 26.7174C50.3333 25.7655 50.6821 24.7428 50.8581 23.6523C50.8046 23.643 50.7701 23.6338 50.7324 23.6338C48.3536 23.5784 46.0534 23.9418 43.8695 24.8937V24.8876ZM36.4474 13.8903C38.0499 13.268 39.6965 13.2803 41.296 13.9334C42.754 14.5279 43.6056 15.8618 43.5804 17.4236H33.8675C34.2037 15.8618 34.8511 14.5125 36.4505 13.8903H36.4474Z"
              fill="#33EFEF"
            />
            <path
              d="M116.14 18.8331C115.688 18.6051 115.217 18.3802 114.726 18.257C113.13 17.8627 111.521 17.5116 109.916 17.1511C109.073 16.9601 108.209 16.8431 107.43 16.4457C107.069 16.2609 106.748 16.0329 106.547 15.6725C106.079 14.8285 106.456 13.6856 107.408 13.2297C108.514 12.7029 109.645 12.6875 110.717 13.3406C112.407 14.3664 114.208 15.1119 116.188 15.3614C116.967 15.46 117.762 15.4692 118.573 15.5216C118.573 15.3337 118.579 15.2166 118.573 15.0995C118.416 12.438 117.174 10.408 114.833 9.04948C113.25 8.12842 111.502 7.76492 109.677 7.72488C107.772 7.68483 105.918 7.92819 104.159 8.6675C102.321 9.4407 100.872 10.6328 100.086 12.4842C99.4296 14.0306 99.4076 15.6263 99.7627 17.2405C100.096 18.7591 100.957 19.9112 102.308 20.7214C103.084 21.1866 103.917 21.5377 104.797 21.7441C106.534 22.1538 108.288 22.505 110.026 22.9116C110.572 23.041 111.106 23.2412 111.622 23.463C111.967 23.614 112.2 23.9159 112.316 24.2855C112.549 25.0218 112.231 25.795 111.512 26.2077C110.908 26.5558 110.239 26.6606 109.554 26.6883C108.605 26.7283 107.669 26.6698 106.82 26.1862C104.922 25.1018 102.896 24.381 100.696 24.1962C100.061 24.1438 99.4202 24.1869 98.7697 24.1869C98.7697 24.2486 98.7571 24.3225 98.7603 24.3933C98.8325 26.5219 99.6653 28.3055 101.331 29.6887C102.833 30.9394 104.621 31.5493 106.544 31.8327C106.99 31.8974 107.443 31.9436 107.892 31.999H109.796C109.875 31.9836 109.953 31.9559 110.032 31.9497C111.352 31.8635 112.637 31.6294 113.875 31.1735C115.239 30.6714 116.461 29.9505 117.423 28.8631C118.947 27.1442 119.402 25.1265 119.107 22.9147C118.855 21.0449 117.847 19.6894 116.14 18.8331Z"
              fill="#33EFEF"
            />
            <path
              d="M22.6593 13.1906C21.8674 10.8464 20.3717 9.13362 17.9647 8.2834C16.0196 7.59646 14.0211 7.54717 12.0163 7.97844C10.1623 8.37582 8.5597 9.21371 7.36247 10.7047C7.3499 10.7201 7.31533 10.7201 7.25877 10.7385V0C7.01367 0.0184829 6.79999 0.0308048 6.58946 0.0462072C2.90665 0.357335 0.00314233 3.44705 0 7.0697C0 15.0789 0 23.0882 0 31.0943V31.387H7.23678V31.0297C7.23678 26.9295 7.23049 22.8294 7.23678 18.7293C7.23678 17.4324 7.617 16.2618 8.59426 15.3346C9.43955 14.5306 10.4954 14.2256 11.6423 14.1486C13.2543 14.0408 14.6967 14.8356 15.4257 16.2464C15.8499 17.0689 16.0164 17.953 16.0164 18.8648C16.0164 22.9188 16.0164 26.9727 16.0164 31.0266V31.3839H23.2532V31.0943C23.2532 26.4551 23.2595 21.8159 23.2501 17.1767C23.2501 15.8213 23.0898 14.4813 22.6499 13.1875L22.6593 13.1906Z"
              fill="#33EFEF"
            />
            <path
              d="M96.7936 18.7385C96.7716 18.3842 96.7213 18.0299 96.6459 17.6819C95.634 13.0211 92.7997 9.96833 88.1616 8.55131C86.2762 7.97526 84.3374 8.01531 82.4017 8.27715C79.935 8.60984 77.8799 9.68493 76.2993 11.5887C75.1838 12.9318 74.3511 14.4289 73.7572 16.0584C73.3078 17.2968 73.025 18.569 73.0439 19.8875C73.069 21.7111 73.5058 23.4454 74.3731 25.0658C75.5137 27.1913 77.0943 28.8825 79.3756 29.8836C81.3333 30.7462 83.3821 31.239 85.5189 31.3931C87.8096 31.5563 89.8679 30.9649 91.6873 29.5848C92.718 28.8024 93.5978 27.8782 94.4117 26.8832C95.037 26.1193 95.6246 25.3368 96.0363 24.4343C96.6836 23.0203 96.8313 21.5201 96.8061 19.9984C96.8061 19.5794 96.825 19.1574 96.7998 18.7415L96.7936 18.7385ZM88.856 20.8517C87.9636 22.5829 86.6973 23.9753 84.8684 24.807C84.0577 25.1767 83.1967 25.3584 82.3137 25.1243C81.0694 24.7947 80.4723 23.8983 80.2492 22.7184C80.1958 22.4412 80.1895 22.1578 80.1707 21.9514C80.2304 20.1216 80.8965 18.5906 81.9523 17.2075C82.8322 16.0554 83.91 15.1312 85.2989 14.6014C86.006 14.3334 86.735 14.2071 87.4891 14.3765C88.6487 14.6353 89.4657 15.584 89.6448 16.8594C89.8459 18.2795 89.5002 19.601 88.8529 20.8517H88.856Z"
              fill="#33EFEF"
            />
            <path
              d="M66.8089 8.21954C64.7192 8.52143 63.1292 9.54415 62.1174 11.3801C62.0483 11.5033 61.9854 11.6266 61.9194 11.7498C61.9037 11.7467 61.8849 11.7405 61.8692 11.7374C61.7497 10.6624 61.6303 9.58419 61.5078 8.50911H54.7109V31.4248C55.1666 31.3755 55.5939 31.3478 56.015 31.28C58.9656 30.7994 61.4167 28.3505 61.8314 25.4425C61.9194 24.8326 61.91 24.2072 61.9351 23.588C61.9823 22.3312 61.9571 21.0682 62.0765 19.8175C62.1834 18.6716 62.6453 17.6304 63.4812 16.7833C64.5181 15.7359 65.8411 15.3016 67.274 15.1937C68.3392 15.1106 69.417 15.1383 70.4886 15.1198C70.5954 15.1198 70.7054 15.1198 70.8216 15.1198V8.45366C69.4893 8.15793 68.1601 8.03163 66.8152 8.2257L66.8089 8.21954Z"
              fill="#33EFEF"
            />
          </g>
          <defs>
            <clippath
              id="clip0_7538_85644"
            >
              <rect
                fill="white"
                height="32"
                width="153.6"
              />
            </clippath>
          </defs>
        </svg>
        <div
          class="css-1wox7bx"
        >
          <p
            class="chakra-text css-ab16yn"
          >
            Are you joining as a fan or an athlete?
          </p>
          <div
            class="css-d24jcg"
          >
            <a
              href="/fan/sign-up"
            >
              <button
                class="chakra-button css-1rcs978"
                type="button"
              >
                Fan
              </button>
            </a>
            <p
              class="chakra-text css-17wkocg"
            >
              As a fan you can follow & support your favorite athlete(s) directly.
            </p>
            <a
              href="/athlete/sign-up"
            >
              <button
                class="chakra-button css-1rcs978"
                type="button"
              >
                Athlete
              </button>
            </a>
            <p
              class="chakra-text css-dbez77"
            >
              As an athlete you can share your stories, thoughts,... with your fans and earn money while doing so.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
